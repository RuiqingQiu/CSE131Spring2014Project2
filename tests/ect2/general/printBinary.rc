function : void printBinary(int h){
   if (h < 0){
      cout << 1;
   }
   else{
      cout << 0;
   }
   int mask = 0x40000000;
   while (mask > 0){
      cout << (int)((mask & h) != 0);
      mask = mask / 2;
   }

   cout << endl;
}

function : int main(){
   printBinary(0x12);
   printBinary(0xABCDEF);
   printBinary(012345);
   printBinary(-1);
   printBinary(-2);
   printBinary(-0xF);
   printBinary(0x70ABCDEF);
   return 0;
}
