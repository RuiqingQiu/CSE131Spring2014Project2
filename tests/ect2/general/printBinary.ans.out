00000000000000000000000000010010
00000000101010111100110111101111
00000000000000000001010011100101
11111111111111111111111111111111
11111111111111111111111111111110
11111111111111111111111111110001
01110000101010111100110111101111
